pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -901,32 +1031,31 @@ func shouldCompressResponse(request, response *dns.Msg) bool {
 	return false
 }
 
-func GetSelectorRegexMap(l7 policy.L7DataMap) (CachedSelectorREEntry, error) {
-	newRE := make(CachedSelectorREEntry)
-	for selector, l7Rules := range l7 {
-		if l7Rules == nil {
-			l7Rules = &policy.PerSelectorPolicy{L7Rules: api.L7Rules{DNS: []api.PortRuleDNS{{MatchPattern: "*"}}}}
+func GenerateRegexpAndFqdns(l7Rules *policy.PerSelectorPolicy) (string, []string) {
+	if l7Rules == nil {
+		l7Rules = &policy.PerSelectorPolicy{L7Rules: api.L7Rules{DNS: []api.PortRuleDNS{{MatchPattern: "*"}}}}
+	}
+	// TODO figure out the len here
+	reStrings := make([]string, 0, len(l7Rules.DNS)/2)
+	fqdnStrings := make([]string, 0, len(l7Rules.DNS)/2)
```

Comment:
If there's no way to know beforehand, then I understand the rationale behind this code and we can probably leave it as is. Not sure we can do better :shrug:

Reply:
The only "easy" way to do this would be in case we could guarantee that each rule has _only_ a `matchPattern` OR a `matchName`. In that way we can have one one with the full length and insert from each end; but not sure if thats going to be clean or not...

Reply:
Agreed, would be a good followup.